metricbeat/internal/metrics/cpu/metrics_procfs_common.go

```diff
@@ -39,8 +41,64 @@ func Get(procfs resolve.Resolver) (CPUMetrics, error) {
 		return CPUMetrics{}, errors.Wrapf(err, "error opening file %s", path)
 	}
 
-	return scanStatFile(bufio.NewScanner(fd))
+	metrics, err := scanStatFile(bufio.NewScanner(fd))
+	if err != nil {
+		return CPUMetrics{}, fmt.Errorf("scanning stat file: %w", err)
+	}
+
+	infoPath := procfs.ResolveHostFS("/proc/cpuinfo")
+	infoFd, err := os.Open(infoPath)
+	if err != nil {
+		return CPUMetrics{}, fmt.Errorf("opening /proc/cpuinfo: %w", err)
+	}
+	defer infoFd.Close()
+
+	cpuInfo, err := scanCPUInfoFile(bufio.NewScanner(infoFd))
+	metrics.CPUinfo = cpuInfo
+
+	return metrics, err
+
+}
+
+func cpuinfoScanner(scanner *bufio.Scanner) ([]CPUInfo, error) {
+	cpuInfos := []CPUInfo{}
+	current := CPUInfo{}
+	// On my tests the same order the cores appear on /proc/cpuinfo
+	// is the same in which they appear on /proc/stats, this means it
+	// matches our current 'system.core.id' metric. This information
+	// is also the same as the 'processor' line on /proc/cpuinfo.
+	coreID := 0
+	for scanner.Scan() {
+		line := scanner.Text()
+		split := strings.Split(line, ":")
+		if len(split) != 2 {
+			// its a separation between CPUs
+			// even the last CPU contains one
+			cpuInfos = append(cpuInfos, current)
+			current = CPUInfo{}
+			coreID++
+
+			continue
+		}
+
+		k, v := split[0], split[1]
+		k = strings.TrimSpace(k)
+		v = strings.TrimSpace(v)
+		switch k {
+		case "model":
+			current.ModelNumber = v
```

Comment:
So, these values come from the `cpuid` instruction, and in many applications the `family` and `model` are reported together to ID the CPU, and we might want to do the same here. If you ever see values like `386` or `686` on software builds or stuff like that, that's derived from the `family` value.